firebase-firestore/src/main/java/com/google/firebase/firestore/remote/OnlineStateTracker.java

```diff
@@ -135,30 +97,36 @@ void handleWatchStreamFailure(Status status) {
       // To get to OnlineState.ONLINE, updateState() must have been called which would have reset
       // our heuristics.
       hardAssert(this.watchStreamFailures == 0, "watchStreamFailures must be 0");
-      hardAssert(this.onlineStateTimer == null, "onlineStateTimer must be null");
+      hardAssert(this.connectivityAttemptTimer == null, "connectivityAttemptTimer must be null");
     } else {
       watchStreamFailures++;
+      clearConnectivityAttemptTimer();
       if (watchStreamFailures >= MAX_WATCH_STREAM_FAILURES) {
-        clearOnlineStateTimer();
         logClientOfflineWarningIfNecessary(
             String.format(
                 Locale.ENGLISH,
-                "Connection failed %d times. Most recent error: %s",
-                MAX_WATCH_STREAM_FAILURES,
+                "Backend didn't respond within %d seconds. Most recent error: %s\n",
+                CONNECTIVITY_ATTEMPT_TIMEOUT_MS / 1000,
                 status));
         setAndBroadcastState(OnlineState.OFFLINE);
       }
     }
   }
 
+  void handleWatchStreamStart() {
```

Comment:
Please be wary of the diffs you're introducing. This function previously was above `handleWatchStreamFailure` and had a comment. Please restore it and its comment (and edit the comment to reflect what you changed).